### PR TITLE
Working

### PR DESCRIPTION
**Pull request by tamio96**
- Change method's name (Ex ExportData -> export_data) to fit PEP 8 convention
- Update Readme
- Add test_suite to setup.py

### CODE REVIEW

File: holistics/holistics.py [hunk 1]
```diff
@@ -89,11 +82,25 @@ def DownloadResults(self):
...
+            return err
         
 
-    def ExportData(self,report_id,filters=None, page_size=None, page=None):
-        self.SubmitReport(report_id, filters)
-        self.GetExportResults(self.page['job_id'],page_size,page)
-        return self.DownloadResults()
+    def export_data(self,report_id,filters=None, page_size=None, page=None):
+        res = self.submit_report(report_id, filters)
+        if type(res) == requests.exceptions.HTTPError:
```
Comment (datbth) [inline holistics/holistics.py:94]: not a proper way to handle Exceptions. Exceptions must be handled by try..except..  
If there is an error, the function having the error must raise the Exception, not returning one (see comment for `submit_report` function).  
Then, in this function we can do something like
```python
try:
    res = self.submit_report(report_id, filters)
    res = self.get_export_result(res,page_size,page)
    res = self.download_results()
    return res
except HolisticsException as e:
    # print error message and return 0 just to follow current behaviour. In fact we do not need to catch this exception, we can just let the caller of `export_data` handle it
    print (e.message)
    return 0
```

File: holistics/holistics.py [hunk 2]
```diff
@@ -27,58 +27,51 @@ def __init__(self, api_key, path = None, url = None):
...
-            print (err)
-            return 0
+    def get_url(self, tail_url, params=None):
+        res = requests.get(self.url + tail_url, params = params, headers = self.headers)
+        res.raise_for_status()
         return res
-        
 
-    def SubmitReport (self, report_id, filters=None):
+    def submit_report (self, report_id, filters=None):
```
Comment (datbth) [inline holistics/holistics.py:35]: should be `submit_export`

File: holistics/holistics.py [hunk 3]
```diff
@@ -27,58 +27,51 @@ def __init__(self, api_key, path = None, url = None):
...
+            res = self.get_url(tail_url, filters)
             res = res.json()
             print ("Success")
-            self.page['job_id'] = str(res['job_id'])
-            return 1
-        else:
-            return 0
+            return str(res['job_id'])
+        except requests.exceptions.HTTPError as err:
+            return err
```
Comment (datbth) [inline holistics/holistics.py:44]: should not return error here. Either:
**Raise a new exception with friendlier message**:
```python
raise Exception('Fail to submit export')
```

To improve further, declare a new exception class like `HolisticsException` and do:
```python
raise HolisticsException('Fail to submit export')
```
The new exception class can be as specific as you want to describe the error appropriately

**Or do not catch the exception here, let `export_data` handle it**

File: holistics/holistics.py (hunk 1 above)
Comment (datbth) [reply holistics/holistics.py:94]: If you dont want to wrap error in `HolisticsException`, remove exception catching in all inner functions and do something like:
```python
try:
    res = self.submit_report(report_id, filters)
except requests.exceptions.HTTPError as e:
    # handle error
try:
    res = self.get_export_result(res,page_size,page)
except requests.exceptions.HTTPError as e:
    # handle error
...
```

File: holistics/holistics.py [hunk 4]
```diff
@@ -27,58 +27,51 @@ def __init__(self, api_key, path = None, url = None):
...
             return 1
-        else:
-            return 0
+        except requests.exceptions.HTTPError as err:
+            return err
         
-    def DownloadResults(self):
+    def download_results(self):
         print ("Downloading results... ", end='')
         tail_url = '/exports/download?job_id=' + self.page['job_id']
```
Comment (datbth) [inline holistics/holistics.py:72]: should document where `page['job_id']` comes from

File: holistics/holistics.py [hunk 5]
```diff
@@ -89,11 +82,25 @@ def DownloadResults(self):
...
-        self.SubmitReport(report_id, filters)
-        self.GetExportResults(self.page['job_id'],page_size,page)
-        return self.DownloadResults()
+    def export_data(self,report_id,filters=None, page_size=None, page=None):
+        res = self.submit_report(report_id, filters)
+        if type(res) == requests.exceptions.HTTPError:
+            print ("Fail to submit report")
+            return 0
+        res = self.get_export_result(res,page_size,page)
+        if res != 1:
```
Comment (datbth) [inline holistics/holistics.py:98]: missing error handling for HTTPError

File: holistics/holistics.py [hunk 6]
```diff
@@ -27,58 +27,51 @@ def __init__(self, api_key, path = None, url = None):
...
+        try:
+            res = self.get_url(tail_url, self.page)
             res = res.json()
             while (res['status'] != 'success'):
                 if res['status'] == 'already_existed':
                     self.page['job_id'] = str(res['job_id'])
-                    break
                 if res['status'] == 'failure':
                     print ("Status: Failure")
                     return 0
```
Comment (datbth) [inline holistics/holistics.py:62]: instead of returning `0`, can raise an Exception here.

**Review by datbth — CHANGES_REQUESTED** [review]

**Comment by phiendp** [comment]
@tamio96 if you are using Sublime Text 3, you can try this package: http://damnwidget.github.io/anaconda/. It will help you enforce the PEP-8 style guide easier. The same package can be found on Vscode or other IDEs.

**Pull request closed without merging**